test/optim/test_zeroptimizer.py

```diff
@@ -0,0 +1,469 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
+
+import os
+
+import numpy as np
+import unittest
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from typing import List, Any
+import io
+from torch.optim import ZeROptimizer, SGD
+skip_if_no_cuda = unittest.skipIf(not torch.cuda.is_available(), reason="cuda required")
+
+BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
+DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def _broadcast_object(
```

Comment:
Similar feature is available in the distributed package. Will the API linked below be sufficient?

https://github.com/pytorch/pytorch/blob/83d358da7c9957db6cc6456f69f43a4c2a4d54e4/torch/distributed/distributed_c10d.py#L1486

Reply:
if this covers dicts that's perfect indeed, thanks for the pointer